aggregated-dependency-score_test.go

```diff
@@ -110,3 +131,68 @@ func TestAggregatedTrustworthinessEvaluation(t *testing.T) {
 		}
 	})
 }
+
+func TestCycleHandling(t *testing.T) {
+	trustworthinessByName := map[string]float64{
+		"A": 0.92,
+		"B": 0.94,
+		"C": 0.93,
+	}
+
+	evaluator := trustwhorthinessEvaluator{
+		intrinsic: &testIntrinsicTrustworthinessEvaluator{
+			trustworthinessByName: trustworthinessByName,
+			maxQueryNumber:        1,
+		},
+		deps: &testDependencyResolver{
+			directDependencyNamesByName: map[string][]string{
+				"A": {"B"},
+				"B": {"C"},
+				"C": {"A"},
+			},
+		},
+	}
+
+	firstScore, err := evaluator.evaluate(context.Background(), Package{Name: "A"}, nil)
+	if err != nil {
+		var tooManyQueriesErr *ErrTooManyQueries
+		if errors.As(err, &tooManyQueriesErr) {
+			t.Fatalf("failed to ignore dependency cycle: %v", err)
+		}
+
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	// Note that what we want is *not* to account for each package only once;
+	// see example https://cedricvanrompay.fr/blog/aggregated-dependency-score/#incentivizing-the-removal-of-transitive-dependencies
+	// we are only ignoring cycles
```

Comment:
ignoring cycles will skew the result as you only partly consider dependencies

Reply:
in other words, you do not get a consistent score for packages

Reply:
![image](https://github.com/user-attachments/assets/cc4ab600-9a67-4991-ad97-6307ae2dedd6)

F gets a different score when walking the left arm compared to the right

Reply:
Hmm, good catch. I'll have to take this sort of situation into account in the algorithm, but this can be done later as it's really an exceptional corner case. Also this will most likely lead to packages get near-zero scores anyway, which makes sense here since the algorithm is supposed to "favor shallow dependency graphs".